File: src/main/java/org/goafabric/personservice/adapter/CalleeServiceAdapter.java
package org.goafabric.personservice.adapter;

import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.service.annotation.GetExchange;

//@CircuitBreaker(name = "calleeservice")
//@MyCircuitBreaker(name = "calleeservice")
public interface CalleeServiceAdapter {

    @GetExchange("/callees/sayMyName")
    Callee sayMyName(@RequestParam("name") String name);

    @GetExchange("/callees/sayMyOtherName/{name}")
    Callee sayMyOtherName(@PathVariable("name") String name);

}

File: src/main/kotlin/org/goafabric/personservice/adapter/CalleeServiceAdapter.kt
package org.goafabric.personservice.adapter

import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.service.annotation.GetExchange

@CircuitBreaker(name = "calleeservice")
interface CalleeServiceAdapter {
    @GetExchange("/callees/sayMyName")
    fun sayMyName(@RequestParam("name") name: String?): Callee?

    @GetExchange("/callees/sayMyOtherName/{name}")
    fun sayMyOtherName(@PathVariable("name") name: String?): Callee?
}
